packaging/installer/kickstart-ng.sh

```diff
@@ -0,0 +1,843 @@
+
+# ======================================================================
+# Constants
+
+REPOCONFIG_URL_PREFIX="https://packagecloud.io/netdata/netdata-repoconfig/packages"
+REPOCONFIG_VERSION="1-1"
+PATH="${PATH}:/usr/local/bin:/usr/local/sbin"
+
+# ======================================================================
+# Defaults for environment variables
+
+RELEASE_CHANNEL="nightly"
+NETDATA_CLAIM_URL="https://app.netdata.cloud"
+NETDATA_USER_CONFIG_DIR="/etc/netdata"
+NETDATA_AUTO_UPDATES="1"
+NETDATA_ONLY_NATIVE=0
+NETDATA_ONLY_STATIC=0
+
+NETDATA_DISABLE_TELEMETRY="${DO_NOT_TRACK:-0}"
+NETDATA_TARBALL_BASEURL="${NETDATA_TARBALL_BASEURL:-https://storage.googleapis.com/netdata-nightlies}"
+
+if [ ! -t 1 ]; then
+  INTERACTIVE=0
+else
+  INTERACTIVE=1
+fi
+
+# ======================================================================
+# Usage info
+
+usage() {
+  cat << HEREDOC
+USAGE: kickstart.sh [options]
+       where options include:
+
+  --dont-wait                Do not prompt for user input. (default: prompt if there is a controlling terminal)
+  --non-interactive          Do not prompt for user input.
+  --dont-start-it            Do not start the agent by default (only for static installs or local builds)
+  --interactive              Prompt for user input even if there is no controlling terminal.
+  --stable-channel           Install a stable version instead of a nightly build (default: install a nightly build)
+  --nightly-channel          Install a nightly build instead of a stable version
+  --no-updates               Do not enable automatic updates (default: enable automatic updates)
+  --auto-update              Enable automatic updates.
+  --disable-telemetry        Opt-out of anonymous statistics.
+  --native-only              Only install if native binary packages are available.
+  --static-only              Only install if a static build is available.
+  --reinstall                Explicitly reinstall instead of updating any existing install.
+  --claim-token              Use a specified token for claiming to Netdata Cloud.
+  --claim-rooms              When claiming, add the node to the specified rooms.
+  --claim-only               If there is an existing install, only try to claim it, not update it.
+  --claim-*                  Specify other options for the claiming script.
+  --no-cleanup               Don't do any cleanup steps. This is intended to help with debugging the installer.
+
+Additionally, this script may use the following environment variables:
+
+  TMPDIR:                    Used to specify where to put temporary files. On most systems, the default we select
+                             automatically should be fine. The user running the script needs to both be able to
+                             write files to the temporary directory, and run files from that location.
+  ROOTCMD:                   Used to specify a command to use to run another command with root privileges if needed. By
```

Comment:
Won't it be better if users notify us about uncommon root commands?

Reply:
The issue here is that they may need additional options to get `sudo` to work correctly. In _most_ cases they shouldn’t, but it costs us nothing to support those cases where they do.